odoo/models.py

```diff
@@ -301,72 +301,6 @@ def _field_create(self):
-                'store': bool(field.store),
-                'copy': bool(field.copy),
-                'related': ".".join(field.related) if field.related else None,
-                'readonly': bool(field.readonly),
-                'required': bool(field.required),
-                'selectable': bool(field.search or field.store),
-                'translate': bool(field.translate),
-                'relation_field': field.inverse_name if field.type == 'one2many' else None,
-                'serialization_field_id': None,
-                'relation_table': field.relation if field.type == 'many2many' else None,
-                'column1': field.column1 if field.type == 'many2many' else None,
-                'column2': field.column2 if field.type == 'many2many' else None,
-            }
-            if getattr(field, 'serialization_field', None):
-                # resolve link to serialization_field if specified by name
-                serialization_field = Fields.search([('model', '=', vals['model']), ('name', '=', field.serialization_field)])
-                if not serialization_field:
-                    raise UserError(_("Serialization field `%s` not found for sparse field `%s`!") % (field.serialization_field, field.name))
-                vals['serialization_field_id'] = serialization_field.id
-
-            if field.name not in cols:
-                query = "INSERT INTO ir_model_fields (%s) VALUES (%s) RETURNING id" % (
-                    ",".join(vals),
-                    ",".join("%%(%s)s" % name for name in vals),
-                )
-                cr.execute(query, vals)
-                field_id = cr.fetchone()[0]
-                self._context['todo'].append((20, Fields.browse(field_id).modified, [list(vals)]))
-
-                module = field._module or self._context.get('module')
-                if module:
-                    xmlid = 'field_%s_%s' % (self._table, field.name)
-                    cr.execute("SELECT name FROM ir_model_data WHERE name=%s", (xmlid,))
-                    if cr.fetchone():
-                        xmlid = xmlid + "_" + str(field_id)
-                    cr.execute(""" INSERT INTO ir_model_data (name, date_init, date_update, module, model, res_id)
-                                   VALUES (%s, (now() at time zone 'UTC'), (now() at time zone 'UTC'), %s, %s, %s) """,
-                               (xmlid, module, 'ir.model.fields', field_id))
-
-            elif not all(cols[field.name][key] == vals[key] for key in vals):
-                names = set(vals) - {'model', 'name'}
-                query = "UPDATE ir_model_fields SET %s WHERE model=%%(model)s AND name=%%(name)s RETURNING id" % (
-                    ",".join("%s=%%(%s)s" % (name, name) for name in names),
-                )
-                cr.execute(query, vals)
-                field_id = cr.fetchone()[0]
-                self._context['todo'].append((20, Fields.browse(field_id).modified, [names]))
-
```

Comment:
Hey, you are removing way too much here!!!

Please remove only what is specific to sparse fields, that is:
 - no need to sort fields before iterating on them (line 311)
 - no attribute `serialization_field_id` (line 330)
 - no check on `serialization_field_id` (lines 335-341)